src/qibo/models/utils.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+
+from qibo import gates as maingates
+from qibo.gates import gates
+from qibo.models.circuit import Circuit
+
+
+def convert_bit_to_energy(hamiltonian, bitstring):
+    """
+    Given a binary string and a hamiltonian, we compute the corresponding energy.
+    """
+    n = len(bitstring)
+    c = Circuit(n)
+    for i in range(n):
+        c.add(gates.X(int(i)))
+    result = c()  # this is an execution result, a quantum state
+    return hamiltonian.expectation(result.state())
+
+
+def convert_state_to_count(state):
+    """
+    This is a function that convert a quantum state to a dictionary keeping track of
+    energy and its frequency.
+    d[energy] records the frequency
+    """
+    m = int(np.log2(state.size))
+    c = Circuit(m)
+    c.add(maingates.M(*[i for i in range(m)]))
+    result = c(state, nshots=100)
+    counts = result.frequencies(binary=True)
+    return counts
+
+
+def compute_cvar(probabilities, values, alpha):
+    """
+    Auxilliary method to computes CVaR for given probabilities, values, and confidence level.
+
+    Args:
+        probabilities (list): list/array of probabilities
+        values (list): list/array of corresponding values
+        alpha (float): confidence level
+
+    Returns:
+        CVaR
+    """
+    sorted_indices = np.argsort(values)
```

Comment:
Are you sure that this should be sorted according to values and not probabilities (from high to low)? Because the loop below has a termination rule according to the probability.

Reply:
yes, we sort according to the values.